dysnix/nifi-registry/values.yaml

```diff
@@ -131,30 +149,102 @@ flowProvider:
     username: nifireg
     password: nifireg
 
-
 # Additional environment variables to set
 extraEnvs: []
 # extraEnvs:
 #   - name: FOO
 #     value: bar
 
-ssh:
-  # Overrides for git over SSH. If you use your own git server, you
-  # will likely need to provide a host key for it in this field.
-  known_hosts: ""
-  config: ""
-    # specify the config which would go in /home/nifi/.ssh/config file
-    # for e.g.
-  # config: |
-  #   Host github.com
-  #   ProxyCommand socat STDIO PROXY:<proxyIP>:%h:%p,proxyport=<proxyPort>,proxyauth=<username:password>
-  #   User git
-  #   Hostname ssh.github.com
-  #   Port 443
-  #   IdentityFile /etc/fluxd/ssh/identity
-
 tests:
   images:
     busybox:
       image: busybox
       tag: 1.33.1
+
+# Configuration to run NiFi Registry securely
+# c.f. https://nifi.apache.org/docs/nifi-registry-docs/html/administration-guide.html#security_configuration
+# TLS keystores and truststores must be configured to run secure;
+# see (e.g.) certManager (below)
+
+security:
+  # Disabled by default (following the principle of least astonishment)
+  enabled: false
+  needClientAuth: true
+  httpsHost: "0.0.0.0"
+  httpsPort: 18443
+  admin: "Initial Administrator"
+  persistence:
+  # storageClass: "-"
+    accessMode: ReadWriteOnce
+    size: 1Gi
+  # ConfigMap with users.xml and authorizations.xml keys; note that these
+  # settings will override the admin: key above if present
+  authConf:
+
+# cert-manager support
+# Setting this true will have cert-manager create a private CA just for NiFi Registry,
+# including certificates for each NiFi Registry node.
+certManager:
```

Comment:
Is it reasonable to request the option to provide an existing `issuerRef` instead of creating & using a self-signed CA to sign the certs that get created?

https://cert-manager.io/docs/usage/certificate/

Reply:
@michael81877 thank you for your review!

Recall that certificate authentication is how NiFi and NiFi Registry not only mutually authenticate each other, but how NiFi Registry accepts proxied user identification for its internal RBAC.  We weren't able to find a clean way of enforcing cert-manager policy to control the issuance of certificates--particularly across namespaces--so fell back to requiring the administrators to manually offer the trusted CA certs to each NiFi and NiFi Registry instance.

To allow our users to access the NiFi and NiFi Registry user interfaces without loading the private CA keys into their browser, we used an Ingress with a certificate signed by our organizational CA.  NGINX ignores TLS errors by default, so it effectively masked the fact that NiFi and NiFi Registry were using private, self-signed CA certificates.

Not opposed if someone wanted to add that support, but it was outside the use cases we ended up with ourselves.

Reply:
Fair enough! That may be something i look to add. Thanks for adding TLS support :)